include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1215 @@
+
+    // field::OFFSET
+    requires std::UnsignedIntegral<typename t::offset_type>;
+
+    // field::REF_OFFSET
+    requires std::UnsignedIntegral<typename t::ref_offset_type>;
+
+    // field::FLAG
+    requires std::UnsignedIntegral<typename t::flag_type>;
+
+    // field::MAPQ
+    requires std::UnsignedIntegral<typename t::mapq_type>;
+
+    // field::EVALUE
+    requires Arithmetic<typename t::e_value_type>;
+
+    // field::BITSCORE
+    requires Arithmetic<typename t::bitscore_type>;
+
+    // field::ALIGNMENT
+    // You can only specify the container type of the query sequence which is either a sequence container type, like
+    // std::vector, resolving to container_type<gapped<alphabet_type>> or a gap_decorator like the
+    // seqan3::gap_decorator_anchor_set which then resolves to seqan3::gap_decorator_anchor_set<sequence_type>
+    // The complete alignment type is a std::pair over the aligned query sequence type described before at the second
+     // position and the aligned ref type, which is either a std::vector<gapped<alphabet_type>>, if the reference
+    // information is given, or a dummy sequence decorated with gaps (ranges::view::repeat_n | detail::access_restrictor_fn).
+    requires sequence_container_concept<
+                 typename t::template alignment_query_container<gapped<typename t::sequence_alphabet>>> ||
+             aligned_sequence_concept<
+                 typename t::template alignment_query_container<
+                     typename t::template sequence_container<typename t::sequence_alphabet>>>;
+
+    requires sequence_container_concept<
+                 typename t::template alignment_ref_container<gapped<typename t::sequence_alphabet>>> ||
+             aligned_sequence_concept<
+                 typename t::template alignment_ref_container<
+                     typename t::template sequence_container<typename t::sequence_alphabet>>>;
+    // field::MATE
+    // the mate type cannot be configured as it will be set to
+    // std::tuple<ref_id_container<ref_id_alphabet>, ref_offset_type, int32_t>
+};
+//!\endcond
+
+// ---------------------------------------------------------------------------------------------------------------------
+// alignment_file_input_default_traits
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The default traits for seqan3::alignment_file_input
+ * \implements AlignmentFileInputTraits
+ * \ingroup alignment
+ *
+ * \details
+ *
+ * If you wish to change a single or a few types from the default, just inherit from this class and
+ * "overwrite" the respective type definitions.
+ *
+ * This example will make the file read into a smaller alphabet and a compressed container:
+ *
+ * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp my_traits
+ */
+struct alignment_file_input_default_traits_sam
```

Comment:
this is still open and substantial

Reply:
yes I did not resolve this?! I just needed some time to think and did other things beforehand.

The default types are all set to the BAM defaults (because SAM does not specify exact types anyway). I agree that the default should be able to values independent of the format. I will just call the struct `alignment_file_input_default_traits` then?
or maybe `dna` and `aa` like in sequence file since in BLAST you often have protein sequences?

Do you propose to parse the values into temporary values (with certain types) in general and then assign them to the variables?